creator-node/src/services/sync/primarySyncFromSecondary.js

```diff
@@ -0,0 +1,463 @@
+const axios = require('axios')
+const _ = require('lodash')
+
+const config = require('../../config')
+const redis = require('../../redis')
+const { WalletWriteLock } = redis
+const models = require('../../models')
+const { logger: genericLogger } = require('../../logging')
+const DBManager = require('../../dbManager')
+const { getCreatorNodeEndpoints } = require('../../middlewares')
+const { saveFileForMultihashToFS } = require('../../fileManager')
+const SyncHistoryAggregator = require('../../snapbackSM/syncHistoryAggregator')
+const { serviceRegistry } = require('../../serviceRegistry')
+const { asyncRetry } = require('../../utils')
+
+const EXPORT_REQ_TIMEOUT_MS = 10000 // 10000ms = 10s
+const EXPORT_REQ_MAX_RETRIES = 3
+const DEFAULT_LOG_CONTEXT = {}
+
+/**
+ * Export data for user from secondary and save locally, until complete
+ * Should never error, instead return errorObj, else null
+ */
+module.exports = async function primarySyncFromSecondary({
+  wallet,
+  secondary,
+  logContext = DEFAULT_LOG_CONTEXT
+}) {
+  await serviceRegistry.initLibs()
+
+  // This is used only for logging record endpoint of requesting node
+  const selfEndpoint = config.get('creatorNodeEndpoint') || null
+
+  const logPrefix = `[primarySyncFromSecondary][Wallet: ${wallet}][Secondary: ${secondary}]`
+  const logger = genericLogger.child(logContext)
+  logger.info(`[primarySyncFromSecondary] [Wallet: ${wallet}] Beginning...`)
+  const start = Date.now()
+
+  // object to track if the function errored, returned at the end of the function
+  let error = null
+
+  try {
+    await WalletWriteLock.acquire(
+      wallet,
+      WalletWriteLock.VALID_ACQUIRERS.PrimarySyncFromSecondary
+    )
```

Comment:
will change this logic as needed once https://github.com/AudiusProject/audius-protocol/pull/3415 ships